internal/util/util.go

```diff
@@ -219,7 +219,61 @@ func FSize(f *os.File) int64 {
 // IsWordChar returns whether or not the string is a 'word character'
 // Word characters are defined as numbers, letters, or '_'
 func IsWordChar(r rune) bool {
-	return unicode.IsLetter(r) || unicode.IsNumber(r) || r == '_'
+	return IsAlphanumeric(r) || IsWordDelimiter(r)
+}
+
+// IsUpperWordChar returns whether or not the string is a 'upper word character'
+// Upper word characters are defined as numbers, upper-case letters, or delimiters ([A-Z0-9_])
+func IsUpperWordChar(r rune) bool {
+	return IsUpperAlphanumeric(r) || IsWordDelimiter(r)
+}
+
+// IsLowerWordChar returns whether or not the string is a 'lower word character'
+// Lower word characters are defined as numbers, upper-case letters, or delimiters ([a-z0-9_])
+func IsLowerWordChar(r rune) bool {
+	return IsLowerAlphanumeric(r) || IsWordDelimiter(r)
+}
+
+// IsWordDelimiter returns whether or not the string is a 'delimiter character'
+// Delimiter characters are defined as '_'
+func IsWordDelimiter(r rune) bool {
```

Comment:
IMHO this function is redundant and actually makes the code less readable (what is "word delimiter"? isn't it rather a "subword delimiter"? etc). What's wrong with using simple explicit `== '_'` checks instead?

Reply:
Some editors allow the user definition of delimiters. In _zsh_ this is realized as a global config option:

```
set-option -g word-separators "_-()@.,[]{}:=|/"
```

There are already quite some references to this method. So, I thought it is better to define the word delimiters just once in one place other than spreading them all over the code. And it would allow a simple feature in future: User defined delimiters coming from the `settings.json`

Reply:
> And it would allow a simple feature in future: User defined delimiters coming from the `settings.json`

Sounds interesting. But then the question arises: would this future feature work with word movements or with sub-word movements? Or perhaps with both (e.g. with 2 separate options)?

So now I'm thinking: ok, maybe let's keep this function, but rename it to `IsSubwordDelimiter()`? After all, that is exactly what it is. So that it won't be confusing anymore, and also would be consistent with that hypothetical future feature.

Reply:
What happens if you split a word? You get sub-words. So, for me it is a word delimiter ...

It certainly has a side-effect on the double-click word selection. Languages using a lot of dashes (`-`)  for variable or function names (like dashes in lisp) could make use of that feature. The problem with my own example is, that this feature should be language dependent. So, those word delimiters could be defined per language. E.g. the delimiter `-` makes just no sense for _python_, but it does for _lisp_.

Reply:
> What happens if you split a word? You get sub-words. So, for me it is a word delimiter ...

You get sub-words, and you delimit them (separate them from each other).

A word delimiter is a character that delimits words from each other, e.g. a whitespace...

> Languages using a lot of dashes (`-`) for variable or function names (like dashes in lisp) could make use of that feature. The problem with my own example is, that this feature should be language dependent. So, those word delimiters could be defined per language. E.g. the delimiter `-` makes just no sense for _python_, but it does for _lisp_.

I think that's not a problem: micro allows setting options per filetype.

We could also add this stuff to the micro's `ftoptions` plugin...

Reply:
Okay, fair enough. It is a `IsSubwordDelimiter`.

> I think that's not a problem: micro allows setting options per filetype.

Ahh, cool. Didn't know about this. Amazing, sounds good to me.